_posts/en/newsletters/2019-05-21-newsletter.md

```diff
@@ -0,0 +1,221 @@
+  functionality previously described by [BIP118][] with modifications to
+  work with Taproot and to reduce the risk of accidental misuse.  One of
+  the signature hashing (sighash) modes available with bip-anyprevout is
+  directly compatible with the proposed [Eltoo][] layer for LN,
+  requiring only modifications for Taproot and the addition of a
+  *chaperone signature* (described later).  A second sighash mode
+  commits to more data than necessary for Eltoo in a way that may make
+  it useful for atypical commitments in Eltoo or for use in other protocols.
+
+    A significant advantage of this proposal over BIP118 noinput is that
+    it can make use of Taproot cooperative spends, allowing the two or
+    more parties to an LN channel or other contract protocol to spend
+    their funds without revealing any of the contract terms (including
+    that anyprevout was in use).  However, if they do need to reveal the
+    terms, any party to the terms can always do that.
+
+    For a quick look at anyprevout, we consider it in the context of a
+    two-party Eltoo-based LN channel.  As a reminder, the key feature of
+    Eltoo is that each balance change in a channel (state update) can be
+    spent by any later state update, so there's no risk of publishing an
+    old state to the block chain like there is with the current
+    penalty-based LN channels.  Eltoo calls this capability *rebinding*
+    and BIP118 proposed to make rebinding available by allowing
+    signatures to skip committing to the input part of the spending
+    transaction---allowing anyone to modify that part of the
+    transaction to bind any later state they knew about.
+
+    The `SIGHASH_ANYPREVOUTANYSCRIPT` mode (any previous output, any
+    script) defined by bip-anyprevout works similarly to BIP118 noinput,
+    with the following changes.  To use anyprevoutanyscript, the public
+    key to which a signature is compared will need to use a special
+    prefix (0x00 or 0x01; not to be confused with the pubkeys used for
+    [bip-taproot's][bip-taproot] output key that uses the same prefix in
+    a different context).  Additionally, the script being evaluated will
+    need to contain at least one signature that doesn't use
+    anyprevoutanyscript or the other new sighash mode described later.
+    This non-anyprevout signature is called the *chaperone signature*
+    because, under reasonable assumptions, it prevents an anyprevout
+    signature from being misused.  (See [Newsletter #34][] for details
+    about the replay problem.)  With the correct prefix and a chaperone
+    signature, anyprevoutanyscript allows the signature to skip
+    committing to the identifier for output being spent (the outpoint),
+    that previous output's scriptPubKey, and the hash of the executed
+    taproot leaf (tapleaf).  The signature still commits to other
+    details about the prevout, such as its bitcoin value.
```

Comment:
Just to make sure I understand, is it the case that ANYPREVOUTANYSCRIPT does not commit to a specific pubkey in the output to be used, but ANYPREVOUT does?

Reply:
Not the pubkey.  It's the scriptPubKey (output script), the thing that specifies the conditions that must be fulfilled in order for the output to be spent.  Imagine you have two outputs each with different (but similar) scripts:

    <600000> OP_CLTV OP_DROP <A pubkey> OP_CHECKSIGVERIFY <Chaperone pubkey> OP_CHECKSIG
    <601000> OP_CLTV OP_DROP <A pubkey> OP_CHECKSIGVERIFY <Chaperone pubkey> OP_CHECKSIG

With anyprevoutanyscript, the same signature by A could spend either of those outputs (assuming it committed to a spend with an nLockTime over 601,000); with anyprevout, the signature could only spend whichever of those output scipts it explicitly committed to.

Reply:
Then, shouldn't we say "commits to other details about the prevout, such as its bitcoin value and pubkey."?

Reply:
I think that would confuse the issue, as we're discussing the bip-taproot/bip-tapscript signature hashing algorithm here (which does not directly commit to the pubkey), rather than the bip-schnorr verification protocol (which does require the signature commit to the pubkey).  I'm adding some extra words to maybe make that clearer.